### PR TITLE
Change the defult Contract to Counter

### PR DESCRIPTION
**Pull request by Tirthnp**

### CODE REVIEW

File: cli/assets/CounterTemplate.sol [hunk 1]
```diff
@@ -1,4 +1,4 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.13;
 
-contract Contract {}
+contract Counter {}
```
Comment (mattsse) [inline cli/assets/CounterTemplate.sol:4]: this should have a `counter` variable which can be incremented

File: cli/assets/CounterTemplate.t.sol [hunk 2]
```diff
@@ -3,7 +3,7 @@ pragma solidity ^0.8.13;
 
 import "forge-std/Test.sol";
 
-contract ContractTest is Test {
+contract CounterTest is Test {
     function setUp() public {}
 
     function testExample() public {
```
Comment (mattsse) [inline cli/assets/CounterTemplate.t.sol:9]: this should be something like `testIncrement` that calls `Counter.increment` and then uses assertEq to ensure it was successful

**Review by mattsse — CHANGES_REQUESTED** [review]

File: cli/assets/CounterTemplate.t.sol [hunk 3]
```diff
@@ -2,11 +2,22 @@
...
-    function setUp() public {}
+    Counter public counter;
+    function setUp() public {
+       counter = new Counter();
+       counter.setNumber(0);
+    }
+
+    function testIncrement() public {
+        counter.increment();
+        assert(counter.number() == 1);
```
Comment (mattsse) [inline cli/assets/CounterTemplate.t.sol:16]: should use assertEq here

**Comment by Tirthnp** [comment]
@mattsse would the changes makes sense ?

**Review by mattsse — CHANGES_REQUESTED** [review]
@onbjerg I think this is a reasonable example

I wonder if it's worth it to add some docs.
since this is considered a template I don't think it's worth it.

also maybe we want to have only 1 test because usually these are renamed/deleted anyway. so only having 1 test should be enough?

File: cli/assets/CounterTemplate.t.sol (hunk 3 above)
Comment (Tirthnp) [reply cli/assets/CounterTemplate.t.sol:16]: Should I just keep the testIncrement ?

**Comment by gakonst** [comment]
i kinda dont wanna anchor into the Counter as an example - what's the issue with calling the default contract this name?

**Comment by mds1** [comment]
@gakonst Just curious, why not?

I think counter can be an interesting template, because it leaves room to expand it and have `forge init` include a very simple test suite that demonstrates each test type.

**Comment by mattsse** [comment]
ref https://github.com/foundry-rs/foundry/issues/2451

**Comment by onbjerg** [comment]
i think this is ok, simple enough to resolve #2451, not so big that it will cause a lot of churn over what to add/remove etc

no need to add docs for it although the book does need to be updated since it does reference the default template in some places

**Comment by Tirthnp** [comment]
Just wanted to confirm about what changes I still need to make ? @onbjerg @mattsse

File: cli/assets/CounterTemplate.t.sol [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    }
+
+    function testIncrement() public {
+        counter.increment();
+        assertEq(counter.number(),1);
+    }
+
+    function testSetNumber() public {
+        counter.setNumber(10);
+        assertEq(counter.number(),10);
```
Comment (gakonst) [inline cli/assets/CounterTemplate.t.sol:21]:
```suggestion
    function testSetNumber(uint256 x) public {
        counter.setNumber(x);
        assertEq(counter.number(), x);
```

File: cli/assets/CounterTemplate.t.sol [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+contract CounterTest is Test {
+    Counter public counter;
+    function setUp() public {
+       counter = new Counter();
+       counter.setNumber(0);
+    }
+
+    function testIncrement() public {
+        counter.increment();
+        assertEq(counter.number(),1);
```
Comment (gakonst) [inline cli/assets/CounterTemplate.t.sol:16]:
```suggestion
        assertEq(counter.number(), 1);
```

**Review by gakonst — APPROVED** [review]
OK no strong opinions. I liked the blank template because it doesn't require you to delete any code, but I guess nicer to have it this way.

**Pull request merged**